### PR TITLE
Feature/lives

### PR DESCRIPTION
**Pull request by Yamsha75**
~~Awaiting for #2 to be merged~~

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -57,112 +60,128 @@ Ball = { -- single object
...
 
-    -- check for keys pressed before game setup
-    if love.keyboard.isDown(KeyEnum.LEFT) then
-        Paddle.acceleration:transform(LeftVector)
-    end
-    if love.keyboard.isDown(KeyEnum.RIGHT) then
-        Paddle.acceleration:transform(RightVector)
-    end
+    -- check for keys pressed and not released before resetting
+    if love.keyboard.isDown(KeyEnum.LEFT) then paddle.acceleration:transform(left) end
```
Comment (Nieminik) [inline main.lua:99]: Or
```lua
love.keyboard.isDown(KeyEnum.LEFT) and paddle.acceleration:transform(left)
```

Less keywords but probably also less readable. What do you think?

File: main.lua [hunk 2]
```diff
@@ -57,112 +60,128 @@ Ball = { -- single object
...
     end
 end
 
-function UpdateBall(dt)
+local function updateBall(dt)
     -- calculate ball position after this update
-    local dBallPosition = Ball.velocity:copy() * Ball.dVelocity * dt
-    local newBallPosition = Ball.position + dBallPosition
+    local dBallPosition = ball.velocity:copy() * ball.dVelocity * dt
+    local newBallPosition = ball.position + dBallPosition
```
Comment (Nieminik) [inline main.lua]: Doesn't `__mul` return a new instance already? Do we need to call copy?

File: main.lua (hunk 1 above)
Comment (Yamsha75) [reply main.lua:99]: Looks fine, but I'd rather focus on code readability 😄

File: main.lua (hunk 2 above)
Comment (Yamsha75) [reply main.lua]: `dBallPosition` is a movement vector relative to current position, while `newBallPosition` is an exact position after the movement

Comment (Yamsha75) [reply main.lua]: But yes, `__mul` returns a new instance

Comment (Yamsha75) [reply main.lua]: Oh... didn't undestand. Good point, `copy()` is redundant

Comment (Yamsha75) [reply main.lua]: Fixed in 6fb4e9591fbb6066c6f76274e62d1785c2009b2f

**Pull request merged**